feat(discover): Add new sort options

shruthilayaj opened the pull request:
Added 'Most Popular' and 'Recently Viewed' sort
options to Discover and Dashboards Manager.

shruthilayaj committed:
feat(discover): Add Discover/Dashboard Visits endpoint

Add an endpoint to record each visit to a Dashboard
or Discover Query

shruthilayaj committed:
Add the new sort by options

shruthilayaj committed:
feat(discover): Track Query/Dashboard visits in frontend

Add an API call to track Dashboard and query visits
in the frontend.

shruthilayaj committed:
feat(discover): Add new sort options

Added 'Most Popular' and 'Recently Viewed' sort
options to Discover and Dashboards Manager.

Zylphrex commented on static/app/views/dashboardsV2/view.tsx:
```diff
@@ -23,6 +24,13 @@ type Props = RouteComponentProps<{orgId: string; dashboardId: string}, {}> & {
+    if (params.dashboardId && params.dashboardId !== 'default-overview') {
+      updateDashboardVisit(api, organization.slug, params.dashboardId);
+    }
+  });
```
You'll want to specify a dependency array here to avoid this hook being called on every render.

shruthilayaj replied:
This PR needs to be updated, it's stale 1 sec

shruthilayaj replied:
Updated @Zylphrex And the thing you mentioned was updated in the previous [PR](https://github.com/getsentry/sentry/blob/master/static/app/views/dashboardsV2/view.tsx#L35)

shruthilayaj committed:
merge with master

shruthilayaj committed:
Remove unintended change

Zylphrex commented:
@shruthilayaj Is the intent to go to GA with this immediately?

shruthilayaj commented:
> @shruthilayaj Is the intent to go to GA with this immediately?

Yeah. I'm going to wait maybe till next week to merge this PR so that we have some meaningful data for most popular and last viewed sorts.

Commit:
fix(itunes): Change required state for SMS slightly (#28314)

The iTunes client has intricate state during authentication. While
ideally much of authentication-related logic and states would be 
split off into a dedicated ITunesAuthenticator class, it would still 
contain a lot of iTunes state anyway.

This is an attempt to keep track of state a little differently by 
making an accessor for one of the integral parts of the client's
state, the trusted phone info responsible for asserting the
correctness of the state machine. This simplifies methods that 
use this trusted phone information and makes it easier to keep
track of state. 

In addition to the above benefits, the trusted phone info was 
only used in one place anyway so there were few gains from 
caching that value.

NATIVE-183

Co-authored-by: Betty Da <bda@sentry.io>

markstory committed:
ref(api) Use ScheduledDeletion for projects (#28274)

Update project deletion to use ScheduledDeletion. This subsystem uses
database records to track deletion jobs so that they are easier to retry
on failure.

wmak committed:
chore(discover): Replace arithmetic with top events (#28282)

* chore(discover): Replace arithmetic with top events

- This is a feature flag for the new top events functionality including
  other and more events in the results

evanpurkhiser committed:
ref(js): Some minor api.tsx cleanup (#28252)

billyvg committed:
build(ci): Remove `make` usage in GH Actions (#28318)

The `setup-sentry` action is used in other repos (e.g. snuba). Makefile will not be available in those repos. Instead, have the action source `scripts/lib` and call `upgrade_pip` directly.

evanpurkhiser committed:
ref(js): Rename StyledSearchBar -> TraceSearchBar (#28321)

wedamija committed:
fix(dev): Cleanup `SENTRY_DEVSERVICES` code to only expect a callable (#27945)

Followup to https://github.com/getsentry/sentry/pull/27943. This removes support for
`SENTRY_DEVSERVICES` entries being just a dict, and now requires them to be a callable.

Depends on
https://github.com/getsentry/sentry/pull/27943
https://github.com/getsentry/getsentry/pull/6036

evanpurkhiser committed:
ref(rrweb): Improve player styling (#28331)

billyvg committed:
build(ci): Fix specifying python-version in `setup-sentry` (#28329)

`python-version` input needs to be piped from `setup-sentry` to `setup-python` to support custom defined python version

ahmedetefy committed:
fix(email): Fix bug that sent emails to deleted emails (#28237)

* fix(email): Fix bug that sent emails to deleted emails

Adds functionality that deletes UserOption instances when the email
is deleted. Also, adds functionality that checks when sending an email
if the UserOption email was deleted, it ignores it and deletes the
`UserOption` instance

Zylphrex committed:
ref(performance): Move transaction overview into folder (#28319)

As more tabs are added to the transaction summary, this folder is getting
polluted with a mix of code from the transaction overview and shared utilities.
This moves the transaction overview into its own folder similar to other tabs.

markstory committed:
feat(deletions) Add metric to validate unused signal (#28324)

I'm pretty sure the pending_delete signal is never called for
repositories, and if that is true we can delete a bunch of code in
repository integrations as it is also unused.

markstory committed:
feat(deletions) Add daily re-attempt for deletions (#28316)

With scheduled deletions we have a record of the work we wanted to do
which is left behind when we can't do the intended work. Because
starting a deletion moves a job into `in_progress` we need to revert
that state on dropped jobs, so that they can be resumed.

I've also fixed the problem where a deletion job that had additional
chunks would never finish because the first chunk would delete the
deletion record which we need to finish the other chunks.

priscilawebdev committed:
ref(grouping-tab): Add project write access condition (#28341)

k-fish committed:
fix(trends): Header overlapping with graph (#28342)

Fixes header overlapping graph due to padding.

scttcper committed:
test(ui): Convert snapshot component tests to RTL (#28310)

edwardgou-sentry committed:
feat(discover): add dashboard widget from discover queries (#28090)

Added Add to Dashboard button in Discover Results page allowing users to create Dashboard widgets from Discover

edwardgou-sentry committed:
feat(discover): added add to discover button to saved discover queries (#28158)

Added Add to Discover Button to Discover landing page in prebuilt/saved queries context menu

armenzg committed:
feat(dev_env): Dev env bootstrap support for two Python versions (#28213)

This change adds supports for bootstrapping two different Python versions. Python 3.8 will be used for Apple M1 since it's the next Python version for Sentry and it is the first one that supports Apple's M1 chip-set.

Other miscellaneous fixes and changes included.

Co-authored-by: Billy Vong <billyvg@users.noreply.github.com>

Zylphrex committed:
ref(performance): Separate header from content in transaction vitals (#28343)

In preparation to unify some of the page layouts in transaction summary pages,
this change separates the header from the content in the transaction vitals tab.

mikellykels committed:
fix(alerts): Project breadcrumb link to project rule list (#28339)

The Project in the breadcrumbs previously went to project settings. This will now go to the list of Alerts for that specific project.

FIXES WOR-1193

joshuarli committed:
fix(django 2.2): rename view method to avoid clashing (#28312)

jmanhart committed:
Adding in new text styles for the codeowners sync CTA in the issues detail. (#28133)

Zylphrex committed:
ref(performance): Separate header from content in transaction events (#28346)

Continuing from #28343, this separates the header from the content on the
transaction events tab.

mgaeta committed:
feat(notifications): Add `actor_id` to Analytics Events (#28347)

relaxolotl committed:
feat(symbol-sources): Redact symbol source secrets from audit logs (#28334)

This redacts any secrets stored in symbol sources before they are logged
as an audit entry.

Co-authored-by: Floris Bruynooghe <flub@sentry.io>

evanpurkhiser committed:
chore(js): Upgrade storybook from 6.3.6 -> 6.3.7 (#28323)

ceorourke committed:
ref(jira ac): Rm fk constraints from JiraTenant (#28349)

mgaeta committed:
types(python): Slack Integration Types (#28236)

joshuarli committed:
py38(django 2.2): bump django-crispy-forms to 1.8.1 (#28344)

mgaeta committed:
feat(notifications): Remove NotificationSettings when Slack is uninstalled (#28216)

Commit:
feat(issue-alerts): Add Sentry Apps Alert Rule Action as an Action (#28338)

Objective:
Issue Alert configurations should return the Sentry App alert rule action as one of the actions. If the Sentry App does not have and Alert Rule UI Component defined but isAlertable, then we should return our current response.

evanpurkhiser committed:
ref(rrweb): Upgrade to latest player (#28333)

wedamija committed:
feat(cdc_search): Implement cdc search for `is:unresolved` query. (#28006)

This implements the basics of `CdcPostgresSnubaQueryExecutor` to handle a single query -
`is:unresolved`. This should work correctly for all existing sorts, filtering by projects,
environments, etc. The results should be equivalent to the existing executor, but we can test that
in the `ServiceDelegator` callback.

This is reasonably rough. As we start to implement more features I'll work to factor this into
methods.

evanpurkhiser committed:
ref(ts): Authenticators have names (#28362)

evanpurkhiser committed:
fix(ui): Use theme border for form footer border (#28359)

wedamija committed:
fix(tests): Temporarily disable test for migration 0223 (#28363)

Stopgap until we get https://github.com/getsentry/sentry/pull/28361 to work as expected. Using
`TransactionTestCase` doesn't 100% solve the problem, since it will leave the database in a not
fully migrated state.

evanpurkhiser committed:
ref(js): Add notice to use useApi when applicable (#28365)

evanpurkhiser committed:
fix(js): useKeyPress should define functions in useEffect (#28366)

Otherwise these functions are just redefined every time. Also changing
the targetKey will not work.

priscilawebdev committed:
fix(yarn.lock): Update lock with new changes (#28369)

untitaker committed:
fix(grouping): Actually fix missing tree labels in JS (#28337)

priscilawebdev committed:
fix(mobile-screenshots): Adjust space for large images (#28309)

jjbayer committed:
feat(metrics): Mock implementation for string indexer service (#28207)

Commit:
fix(appconnect): correct docstring (#28336)

matejminar committed:
feat(ui): Add expanding to release comparison (#28340)

priscilawebdev committed:
ref(http-custom-repo): Add clear password button (#28268)

priscilawebdev committed:
ref(custom-repo): Add App Store Connect delete msg (#28368)

JoshFerge committed:
perf(auditlog): add event to grouped datetime index (#28296)

armenzg committed:
feat(devenv): Leniant Python version check (#28373)

In #28213 we removed the leniant Python version check and it hit few engineers.
This adds it back and better wording for Python 3.8 version check.

Commit:
fix(appconnect): Ensure a bool is returned for promptItunesSession (#28375)

This is supposed to return a boolean JSON type, but that is not what
it was doing, if there were no pending downloads it would return `0`
if there was no valid itunes session.

markstory committed:
ref(ui) Convert ProjectDashboard container to a FC (#28357)


Co-authored-by: Priscila Oliveira <priscilawebdev@gmail.com

armenzg committed:
fix(dev): Typo for Python version check (#28376)

Fixes #28373

evanh committed:
ref: Track time event was added to nodestore (#28313)

Add a timestamp to the event payload representing when the event was added to
nodestore. This is then used to help track down ingest delay failures.

billyvg committed:
fix(devenv): Fix py3 bootstrapping for getsentry (#28377)

This fixes bootstrapping py3 for getsentry, as we need to ensure the proper paths when these scripts are used outside of the current repo.

markstory committed:
ref(ui) Convert AutoSelect to a function component (#28286)

This view required access to the component in the past. With function
components this style of API is created with useImperativeHandle() and
forwardRef().

Zylphrex committed:
ref(performance): Separate header from content in transaction tags (#28354)

Continuing from #28343, this separates the header from the content on the
transaction tags tab.

Commit:
feat(issue-alert): Render Sentry App Alert Rule once selected (#28355)

wmak committed:
feat(discover): Add other to top-events response (#28322)

- This adds other to the top-events response so that the scale of the
  top events can be seen against the whole
    - Done via a third query to keep things simple
    - Construction the negation of each condition as they're being made
      instead of once at the end. The reason for this is to hopefully
      make it easier to create negations if we need to introduce more
      complicated conditions.
- Will add tests in a follow up PR, the changes are behind a feature
  flag, and existing tests continue to pass so this should be a low risk
  change
- Also cleaned up feature flag checking on the endpoint a bit, moving
  interpolation to a class function instead

Zylphrex committed:
ref(performance): Separate header from content in transaction overview (#28378)

Continuing from #28343, this separates the header from the content on the
transaction overview tab.

mgaeta committed:
feat(notifications): Confirmation Modal (#27728)

Co-authored-by: Nisanthan Nanthakumar <nisanthan.nanthakumar@sentry.io>

joshuarli committed:
py38(django 2.2): upgrade django-picklefield to 2.1.0 (#28317)

matejminar committed:
test(ui): Add test coverage of release comparison expanding (#28374)

priscilawebdev committed:
ref(mobile-screenshot): Update dialog preview displayed info (#28392)

matejminar committed:
fix(ui): Fix calculation of relative session duration (#28399)

matejminar committed:
feat(ui): Session Duration all releases comparison (#28401)

matejminar committed:
feat(ui): Rename unadopted to replaced (#28400)

priscilawebdev committed:
feat(tests): Add mobile screenshot tests (#28391)

priscilawebdev committed:
ref(dynamic-sampling): Introduce new design (#28393)

matejminar committed:
fix(ui): Fix count_unique in release comparison (#28409)

shruthilayaj committed:
fix merge conflits

shruthilayaj commented:
Did a bad merge with master, re-opened a new PR [here](https://github.com/getsentry/sentry/pull/28413/files)

untitaker reviewed (commented):
ship it!
